wire-ios-sync-engine/Source/Synchronization/TimerActionsManager.swift

```diff
@@ -0,0 +1,116 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+@objc
+public protocol TimerActionsManagerType: AnyObject {
+
+    func applyTimerActionsIfNeeded(_ checkTime: Date)
+
+}
+
+public final class TimerActionsManager: NSObject, TimerActionsManagerType {
+
+    // MARK: - Models
+
+    struct Event {
+        let action: (() -> Void)?
+    }
+
+    /// Performing actions every 3 hours.
+    private let interval: TimeInterval = 3 * 60 * 60
+
+    weak var managedObjectContext: NSManagedObjectContext?
```

Comment:
Why does this class need to know about a context? Doing so tightly couples it to Core Data. If you need to switch to a context and perform saves, do it within the action instead.

Reply:
removed